Release notes and migration guide ready to merge (#2370)
tinymce/tinymce-docs: modules/ROOT/pages/6.0-release-notes-options.adoc

In {productname} 6, `images_upload_handler` reports upload results by returning a `Promise`, with a value of either `resolve` or `reject`.

[[link-default-protocol]]
==== link_default_protocol

The `link_default_protocol` option sets the link protocol used by links added or edited using the _link_ dialog.

It now defaults to `https`. Previously it defaulted to `http`.

NOTE: The `link_default_protocol` value is only applied to an edited or inserted link if the protocol is not explicitly specified.

[[tinymce-active-editor-options-get]]
==== tinymce.activeEditor.options.get

Previously `tinymce.activeEditor.options.get('plugins')` returned the configured plugins as a string.

As of {productname} 6, `tinymce.activeEditor.options.get('plugins')` now returns the configured plugins as an array.

//=== Fixed options

//=== Added options

[[removed-or-deprecated-options]]
=== Removed or deprecated options

[[textpattern]]
==== TextPattern

* The `textpattern_patterns` option is no longer an option. It is has been renamed, to `text_patterns` and is now a Core part of {productname} 6.0.
+

> All references to `textpattern_patterns` (and `text_patterns`) should be removed from your configurations.

Reviewer: I don't understand the 

> (and `text_patterns`)

part of this sentence as we specifically tell people to rename it to that and then say to remove it?!

+

> * Also, the API called `textpattern` has been removed.

Reviewer: Do we mention that it can be set via the tinymce.options now?

Reviewer 2: I was going to comment on the API being mentioned in the options page at all, but James makes a good point. Maybe: 
```suggestion
* The `textpattern` API has also been removed. The EditorOptions API can be used instead to get and set textpatterns using `editor.options.get('text_pattern')` and `editor.options.set('text_pattern', newPatterns)`.
```

Reviewer 2: As commented elsewhere, the textpattern API wasn't actually documented, so it may be better to just not mention it at all.
```suggestion
```

Reviewer 3: Yeah I'd vote to just remove this, as we don't want to advertise updating the options at runtime too much (which is also why the API wasn't documented).

= TinyMCE 6.0 Options changes
:navtitle: TinyMCE 6.0 Options changes
:description: TinyMCE 6.0 Options changes
:keywords: releasenotes, options, configurations

This section documents the changes made to {productname} 6.0 Options.

It also documents the {productname} 6.0 Option that has been removed.

* xref:options[Options]
* xref:changed-options[Changed Options]
* xref:removed-or-deprecated-apis[Removed or deprecated Options]

// tag::options[]
[[options]]
== Options

[[changed-options]]
=== Changed options

[[image-upload-handler]]
==== `image_upload_handler`

Previously the `images_upload_handler` option passed `success` or `failure` callbacks to report the status of an image upload.

> In {productname} 6, `images_upload_handler` reports upload results by returning a `Promise`, with a value of either `resolve` or `reject`.

Reviewer 4: This is a general point, repeated multiple times in the doc, but your description is not how JavaScript promises work.

Promises can be created in two ways; either resolved with a value or rejected with an error. These are not themselves values of Promise.

To be honest, I think we can just remove the 'with a value' text completely. Anyone who doesn't already know how promises work isn't going to figure them out by us trying to explain that here.
```suggestion
In {productname} 6, `images_upload_handler` reports upload results by returning a `Promise`.
```

- [ ] if this is accepted repeat the change every other time it appears

[[link-default-protocol]]
==== link_default_protocol

The `link_default_protocol` option sets the link protocol used by links added or edited using the _link_ dialog.

It now defaults to `https`. Previously it defaulted to `http`.

NOTE: The `link_default_protocol` value is only applied to an edited or inserted link if the protocol is not explicitly specified.

[[tinymce-active-editor-options-get]]
==== tinymce.activeEditor.options.get

Previously `tinymce.activeEditor.options.get('plugins')` returned the configured plugins as a string.

As of {productname} 6, `tinymce.activeEditor.options.get('plugins')` now returns the configured plugins as an array.

> 

Reviewer 2: I wonder if this should be moved to its own subsection in this page, as it's a change to the Options API (which lets you register, change, etc. options) vs an option itself?

Reviewer 2: I don't mind if we defer or don't do this

= TinyMCE 6.0 Options changes
:navtitle: TinyMCE 6.0 Options changes
:description: TinyMCE 6.0 Options changes
:keywords: releasenotes, options, configurations

This section documents the changes made to {productname} 6.0 Options.

It also documents the {productname} 6.0 Option that has been removed.

* xref:options[Options]
* xref:changed-options[Changed Options]
* xref:removed-or-deprecated-apis[Removed or deprecated Options]

// tag::options[]
[[options]]
== Options

[[changed-options]]
=== Changed options

[[element-format]]
==== `element_format`

> Previously, `element_format` had no explicit default value. It’s implicit value, however, was `xhtml`.

Reviewer 3: ```suggestion
Previously, `element_format` had no explicit default value. It’s implicit value, however, was `'xhtml'`.
```

> In {productname} 6.0, `element_format` now has an explicit default value: `html`.

Reviewer 3: ```suggestion
In {productname} 6.0, `element_format` now has an explicit default value: `'html'`.
```

[[image-upload-handler]]
==== `image_upload_handler`

Previously the `images_upload_handler` option passed `success` or `failure` callbacks to report the status of an image upload.

In {productname} 6, `images_upload_handler` reports upload results by returning a `Promise`, with a value of either `resolve` or `reject`.

[[link-default-protocol]]
==== link_default_protocol

The `link_default_protocol` option sets the link protocol used by links added or edited using the _link_ dialog.

> It now defaults to `https`. Previously it defaulted to `http`.

Reviewer 3: ```suggestion
It now defaults to `'https'`. Previously it defaulted to `'http'`.
```

NOTE: The `link_default_protocol` value is only applied to an edited or inserted link if the protocol is not explicitly specified.

[[tinymce-active-editor-options-get]]
==== tinymce.activeEditor.options.get

> Previously `tinymce.activeEditor.options.get('plugins')` returned the configured plugins as a string.

Reviewer 3: This statement is incorrect, as this API didn't exist before. I'd suggest rephrasing this as something like this and update the title to match?
```suggestion
Previously the `plugins` option returned the configured plugins as a string.
```

> As of {productname} 6, `tinymce.activeEditor.options.get('plugins')` now returns the configured plugins as an array.

Reviewer 3: ```suggestion
As of {productname} 6, the `plugins` option now returns the configured plugins as an array.
```

Reviewer 2: For both this line and the one before it:

Previously, the list of configured plugins could be retrieved by calling `editor.getParam('plugins')` and returned the configured plugins as a string.

As of {productname} 6, the list of configured plugins is retrieved by calling `editor.options.get('plugins')` and now returns the configured plugins as an array.

> //=== Fixed options
> 
> //=== Added options
> 

Reviewer 3: ```suggestion
```
As these don't appear to be used we should probably just remove them.

[[removed-or-deprecated-options]]
=== Removed or deprecated options

[[textpattern]]
==== TextPattern

* The `textpattern_patterns` option is no longer an option. It has been renamed to `text_patterns` and is now a Core part of {productname} 6.0.
+

> All references to `textpattern_patterns` should be removed from configurations.

Reviewer 3: This doesn't seem correct? We want them instead to rename `textpattern_patterns` to `text_patterns`
